### ISSUE TITLE
Focus in an empty DGV

### ISSUE THREAD

**Issue by Tanya-Solyanik** [dotnet/winforms#7026]
@vladimir-krestov  - should this be `return owner.Focused && RowCount == 0;`

_Originally posted by @Tanya-Solyanik in https://github.com/dotnet/winforms/pull/6102#discussion_r851568124_

**Comment by vladimir-krestov** [comment]
I think so, because we have `HasKeyboardFocus` as true, even a DGV doesn't have focus. I think, it's not correct and we should fix it. Will take it

**Comment by Cassie-Li01** [comment]
Verified this issue on .NET 7.0.100-preview.5.22227.7, issue was fixed. 
Without focus, HasKeyboardFocus as false
![image](https://user-images.githubusercontent.com/96758262/165707713-fd6fe515-d793-4e2e-b9d6-bff9d599cff7.png)

With Focus:  HasKeyboardFocus as True
![image](https://user-images.githubusercontent.com/96758262/165707974-d5ba61db-ac8d-41b8-8572-8635aa7bcf6a.png)
